account_statement_line_order/models/account_bank_statement_line.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2025 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from xmlrpc.client import MAXINT
+
+from odoo import api, models
+
+
+class AccountBankStatementLine(models.Model):
+    _inherit = "account.bank.statement.line"
+
+    @api.depends("journal_id.statement_line_internal_index")
+    def _compute_internal_index(self):  # pylint: disable=missing-return
+        """Compute the internal index based on the defined order."""
```

Comment:
Better to not override the docstring of original method? And put that as comment

```suggestion
        # Compute the internal index based on the defined order.
```